core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -138,7 +139,14 @@ class AclAuthorizer extends Authorizer with Logging {
   override def configure(javaConfigs: util.Map[String, _]): Unit = {
     val configs = javaConfigs.asScala
     val props = new java.util.Properties()
-    configs.foreach { case (key, value) => props.put(key, value.toString) }
+    configs.foreach { case (key, value) =>
+      value match {
+        case password: Password => props.put(key, password)
+        case list: util.List[_] => props.put(key, list.asScala.map(_.toString).mkString(","))
+        case clazz: Class[_] => props.put(key, clazz.getName())
+        case _ => props.put(key, value.toString)
+      }
+    }
```

Comment:
Without this lot of test cases failed because this code converted Class type in toString like 'class a.b.c' and then it could not load DefaultSslEngineFactory

Reply:
We have other class configs like `principal.builder.class` which don't need this. Can you check without setting default value for the new config and see if it still causes tests to fail?

Reply:
Checking now. Basically the 'client' package is fine with making default Class value but KafkaConfig having default as Class value is resulting into lot of test failures. Thanks for the pointer about principal.builder.class, let me check.

Reply:
looks like your suggestion of keeping 'null' as default value and then in the code default to the DefaultSslEngineFactory class seems to work. Will soon add test as pointed out in other comment and commit.

Reply:
done